Fix Re-logging breaks player sprites for all equipment. #2647

JesterX666 opened the pull request:
Clothing items were never removed from equipment slots, they were merely hidden, making them reappearing when you re-logging.  Even if you dropped them on the floor!

JesterX666 committed:
Fix Re-logging breaks player sprites for all equipment. #2647

Clothing items were never removed from equipment slots, they were merely hidden, making them reappearing when you re-logging.  Even if you dropped them on the floor!

chairbender commented on UnityProject/Assets/Scripts/Items/ClothingItem.cs:
```diff
@@ -93,6 +93,8 @@ public virtual void SetReference(GameObject Item)
 		UpdateReferenceOffset();
 		if (Item == null)
 		{
+			GameObjectReference = null; // Remove the item from equipment
```
this causes an nre when you remove your clothing after spawning:

```
var unequippedClothing = GameObjectReference.GetComponent<ClothingV2>();
```

Need to ensure this hook is still fired before nulling out GameObjectReference.

JesterX666 replied:
Was the NRE in client or server?

chairbender replied:
Both. I hosted in editor, joined as client, then as client removed my suit and dropped it on the ground.

JesterX666 replied:
Looking at it.  Thanks for finding it.

JesterX666 replied:
Corrected by moving the null assignation bellow where the reference is needed.

chairbender reviewed (changes requested):
getting an NRE

JesterX666 committed:
Correction of a NRE that I caused because the hook was needing the

Correction of a NRE that I caused because the Correction of a NRE that I caused because the hook was needing a valid reference to GameObjectReference.

Moved the NULL assignation after the hook invoke.
